HIP-XXXX.md

```diff
@@ -0,0 +1,298 @@
+
+Traditional light clients do not protect users’ privacy. First, name lookups are leaked to full nodes through
+`getproof` requests. Light clients generally rely on recursive resolvers that are in theory decentralized but
+in practice very unreliable for end users. Recursive resolvers today do plain text DNS lookups leaking
+all DNS traffic to internet service providers. On some networks, it’s not possible to do recursion
+due to interference by middleboxes[0]. Recursion is incompatible with most VPNs. Plain text DNS is easily
+censorable. In fact, Handshake is unintentionally censored on some networks[1]. The internet is moving
+away from plain text DNS by relying on technologies such as DNSCrypt, DoH, or Oblivious DoH to greatly
+improve users’ privacy and enable technologies such as Encrypted Client Hello[2] to hide all traffic from
+eavesdroppers but ECH would be moot without encrypted DNS.
+
+To have any chance of competing with existing Web PKI, Handshake must provide a comparable solution
+in terms of performance and reliability. For a client to verify a site’s identity using
+DNS-Based Authentication, they need to request a proof from a full node assuming a peer is
+available to serve that proof and do multiple DNS lookups to get the DNSSEC authentication chain
+before they’re able to visit a site. This not only introduces undesirable latency it also makes
+browsing the internet very flaky since it relies on all these network lookups to succeed
+just to verify a certificate[3].
+
+## Conventions and Definitions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",
+"NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described
+in RFC2119 RFC8174 when, and only when, they appear in all capitals, as shown here.
+
+## Overview
+
+Handshake has a unique design; it’s a referral only zone meaning it only allows a subset of DNS records
+on-chain such as DS, NS and TXT records. DNS data is stored in a dedicated merkle tree called Urkel
+similar to a sparse merkle tree used by Certificate Transparency. A light client defined by this standard
+only needs to know the last 40 tree roots. They SHOULD be retrieved from the Handshake network and can be fetched
+on demand. The tree root changes every 36 blocks (approximately 6 hours). It SHOULD be updated by clients once a day
+but it MUST be updated at least once a week . This document does not restrict how the tree roots are fetched or updated but they
+MUST be obtained securely.
+
+
+## Theory of operation
+
+This section explains the theory of operation from a client’s perspective.
+
+### Obtaining a tree root:
+
+Clients MUST store the latest 40 tree roots. Clients SHOULD initially download block headers and verify proof of work. Clients SHOULD update the tree roots at least once a day and MUST update once a week. Refer to Timing Considerations for more details.

```

Comment:
It's true that the root zone isn't meant to be updated regularly, but isn't 1 week a bit too long? If I read this correctly, an out of sync (by max 1 week) client will consider stale records as valid for a whole week!
Syncing headers isn't heavy and can be done more often, like daily or every 2 days (prune length).

Reply:
I agree with having the client check in more often. 1 or two days is  fine, weekly is stretching it IMO.

Reply:
With DNSSEC we usually sign records that expire at least two weeks at a time. Accepting records from the last 40 tree roots is acceptable, IMO. It's a good tradeoff to enable server certificates to be renewed weekly instead of daily, making it less of an operational burden. The HIP recommends that clients update once a day. They shouldn't stay off the network for a week unless absolutely necessary. This timing is certainly not final. We should experiment with it once we have an implementation and adjust as needed.